tests/shared/core/test_events.py

```diff
@@ -614,3 +628,143 @@ def test_event_executed_comparison(
+                {"id": 0, "name": "deny", "confidence": 0.011255578137934208,},
+                {"id": 3, "name": "bot_challenge", "confidence": 0.004019865766167641,},
+                {"id": 4, "name": "affirm", "confidence": 0.002524246694520116,},
+                {"id": 5, "name": "mood_great", "confidence": 0.002214624546468258,},
+                {"id": 6, "name": "chitchat", "confidence": 0.0009614597074687481,},
+                {
+                    "id": 7,
+                    "name": "mood_unhappy",
+                    "confidence": 0.00024030178610701114,
+                },
+            ],
+            "response_selector": {
+                "all_retrieval_intents": [],
+                "default": {
+                    "response": {
+                        "id": 11,
+                        "responses": [{"text": "chitchat/ask_name"}],
+                        "response_templates": [{"text": "chitchat/ask_name"}],
+                        "confidence": 0.9618658423423767,
+                        "intent_response_key": "chitchat/ask_name",
+                        "utter_action": "utter_chitchat/ask_name",
+                        "template_name": "utter_chitchat/ask_name",
+                    },
+                    "ranking": [
+                        {
+                            "id": 11,
+                            "confidence": 0.9618658423423767,
+                            "intent_response_key": "chitchat/ask_name",
+                        },
+                        {
+                            "id": 12,
+                            "confidence": 0.03813415765762329,
+                            "intent_response_key": "chitchat/ask_weather",
+                        },
+                    ],
+                },
+            },
+        },
+    ),
+    SessionStarted(),
+    ActionExecuted(action_name="action_listen"),
+    AgentUttered(),
+    EndToEndUserUtterance(),
+    WronglyClassifiedUserUtterance(
+        event=UserUttered(), eval_store=EvaluationStore(intent_targets=["test"])
+    ),
+    WronglyPredictedAction(
+        action_name_prediction="test",
+        action_name_target="demo",
+        action_text_target="example",
+    ),
+    WarningPredictedAction(action_name_prediction="test"),
+]
+
+
+@pytest.mark.parametrize("event", tested_events)
+def test_event_fingerprint_consistency_across_runs(event: Event):
+    f1 = event.fingerprint()
+
+    event2 = copy.copy(event)
+    f2 = event2.fingerprint()
+
+    assert f1 == f2
+
+
+@pytest.mark.parametrize("event_class", rasa.shared.utils.common.all_subclasses(Event))
+def test_event_subclasses_are_tested(event_class: Type[Event]):
+    subclasses = [event.__class__ for event in tested_events]
+    assert event_class in subclasses
+
+
+@pytest.mark.parametrize(
+    "event", tested_events,
+)
+def test_event_fingerprint_uniqueness(event: Event):
+    f1 = event.fingerprint()
+    event.timestamp = 1
```

Comment:
Tobias makes a good point about `timestamp`. Does this mean that when we re-read the same training data file the event fingerprints won't match?

Reply:
I don't know enough of what reading the training data file entails to be able to answer, but if timestamp is different every time the training data file is re-read, I will remove the timestamp from the fingerprint.